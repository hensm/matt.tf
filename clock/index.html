<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8">
        <title>New Tab</title>

        <link href='https://fonts.googleapis.com/css?family=Cabin:600' rel='stylesheet' type='text/css'>
        <link rel="stylesheet" href="style.css">
    </head>
    <body>
        <div id="clock-wrapper">
            <img id="background" class="center x y hidden">
            <video id="video-background" class="center x y hidden" autoplay loop></video>
            <div id="overlay"></div>
            <div id="subreddit" class="no-select">
                <span id="subreddit-name"></span>
                <span id="subreddit-sort"></span>
                <div id="subreddit-choose" class="hidden">
                    <div id="subreddit-choose-inner" class="center y">
                        <div id="subreddit-choose-change">
                            <div>Change</div>
                        </div>
                        <input id="subreddit-choose-name" type="text">
                        <input id="subreddit-choose-sort" type="text">
                        <div class="clearfix"></div>
                    </div>
                </div>
            </div>
            <div id="toggle-sound" class="muted hidden"></div>
            <div id="clock" class="center x y no-select no-click"></div>
        </div>

        <script src="https://ajax.googleapis.com/ajax/libs/swfobject/2.2/swfobject.js"></script>
        <script src="main.js"></script>
    </body>
</html>