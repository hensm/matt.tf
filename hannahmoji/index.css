:root {
    background-color: black;
    color: white;
    font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", Roboto, Helvetica, Arial, sans-serif, "Apple Color Emoji", "Segoe UI Emoji", "Segoe UI Symbol";
    font-size: 10px;
}

html,
body {
    height: 100%;
}

body {
    align-items: center;
    display: flex;
    height: 100%;
    justify-content: center;
    margin: initial;
}

#canvas {
    position: absolute;
}

.container {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: 2rem;
    z-index: 1;
}

.emoji {
    align-items: center;
    display: flex;
    flex-direction: column;
}
.emoji__image {
    filter: drop-shadow(0 0 5rem black);
}
.emoji__title {
    font-size: 2em;
    margin-top: 2rem;
    font-weight: bold;
}
.emoji__description {}

.emoji__title,
.emoji__description {
    text-shadow: 0 0 1rem black;
}


.search {
    margin-top: 4rem;
    width: 100%;
}
.search__input {
    background-color: rgba(255, 255, 255, 0.5);
    border: initial;
    border-radius: 1rem;
    color: inherit;
    font: inherit;
    padding: 1rem 2rem;
    width: 100%;
}

.search__input::placeholder {
    color: black;
}
