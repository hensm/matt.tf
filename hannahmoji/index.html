<html>
    <head>
        <title>Hannahmoji</title>
        <meta charset="utf-8">
        <meta name="viewport" content="width=1000, user-scalable=0">

        <link rel="stylesheet" href="index.css">

        <script src="tagcanvas.min.js" defer></script>
        <script src="index.js" defer></script>
    </head>
    <body>
        <canvas id="canvas" width="500" height="500"></canvas>
        <div class="container">
            <div class="emoji">
                <img width="398" height="398"
                     class="emoji__image"
                     srcset="https://render.bitstrips.com/v2/cpanel/20011475-191405008_41-s5-v1.png?transparent=1&palette=1,
                             https://render.bitstrips.com/v2/cpanel/20011475-191405008_41-s5-v1.png?transparent=1&palette=1&scale=2 2x">
                <div class="emoji__title">
                    I love the internet
                </div>
                <div class="emoji__description">
                    hanging out with a globe that is wearing a sash that says "internet"
                </div>
            </div>
            <div class="search">
                <input type="text" class="search__input" placeholder="Search...">
            </div>
        </div>
    </body>
</html>