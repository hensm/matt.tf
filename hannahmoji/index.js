"use strict";

const AVATAR_ID_HANNAH = "191405008_41-s5";
const AVATAR_ID_ME = "191405008_41-s5";
//const AVATAR_ID_ME     = "519634393_3-s5";


// Elements
const elEmoji            = document.querySelector(".emoji")
const elEmojiImage       = document.querySelector(".emoji__image");
const elEmojiTitle       = document.querySelector(".emoji__title");
const elEmojiDescription = document.querySelector(".emoji__description");
const elSearch           = document.querySelector(".search");
const elSearchInput      = document.querySelector(".search__input");
const elCanvas           = document.querySelector("#canvas");

elSearchInput.value = "";

function fitCanvas () {
    elCanvas.width = window.innerWidth;
    elCanvas.height = window.innerHeight;
}

fitCanvas();

window.addEventListener("resize", fitCanvas);



function displayEmoji (template, avatarId1, avatarId2) {
    const url = template.src
            .replace("%s", avatarId1)
            .replace("%s", avatarId2);

    elEmojiImage.srcset = `
            ${url},
            ${url}&scale=2 2x`;

    elEmojiTitle.textContent = template.alt_text;
    elEmojiDescription.textContent = template.descriptive_alt_text;
}


function getRandomTemplate (templates) {
    return templates[Math.floor(Math.random() * templates.length)];
}


fetch("templates.json")
    .then(res => res.json())
    .then(init)
    .catch(err => {
        console.error(err);
    });

function buildWorker (data, onResults) {
    return new Promise ((resolve, reject) => {
        const worker = new Worker("worker.js");

        worker.postMessage({
            subject: "init"
          , data: data
        });

        worker.addEventListener("message", ev => {
            const message = ev.data;

            switch (message.subject) {
                case "init:response": {
                    resolve(worker);
                    break;
                };
                case "search:response": {
                    onResults(message.data);
                    break;
                }
            }
            if (ev.data.subject === "init:response") {
                resolve(worker);
            }
        });
    });
}

async function init (res) {
    const templates = res.imoji.concat(res.friends);

    // Limit DPI to 2
    const dpi = Math.min(window.devicePixelRatio, 2);


    TagCanvas.Start("canvas", null, {
        imageMode: "image"
      , imageScale: (0.5 / dpi)
      , noTagsMessage: false
      , pinchZoom: true
      , zoom: 1.25
      , reverse: true
      , maxSpeed: 0.025
    });


    function onResults (results) {
        // Clear tags
        while (elCanvas.firstChild) {
            elCanvas.removeChild(elCanvas.firstChild);
        }

        for (const result of results) {
            const url = result.src
                    .replace("%s", AVATAR_ID_HANNAH)
                    .replace("%s", AVATAR_ID_ME);

            const tag = document.createElement("a");
            const img = document.createElement("img");

            img.src = `${url}&scale=${dpi}`;

            tag.addEventListener("click", ev => {
                ev.preventDefault();
                
                displayEmoji(result, AVATAR_ID_HANNAH, AVATAR_ID_ME);
            });

            tag.appendChild(img);
            canvas.appendChild(tag);
        }

        TagCanvas.Reload("canvas");
        TagCanvas.SetSpeed("canvas", [ 0.05, -0.05 ]);
    }

    let worker = await buildWorker(templates, onResults);
    let hasSearched = false;

    let timeout;

    elSearchInput.addEventListener("input", async ev => {
        if (!ev.target.value) {
            return;
        }

        window.clearTimeout(timeout);
        window.setTimeout(async () => {
            if (hasSearched) {
                hasSearched = false;
                worker.terminate();
                worker = await buildWorker(templates, onResults);
            }

            worker.postMessage({
                subject: "search"
              , data: ev.target.value
            });

            hasSearched = true;
        }, 500);
    });
}
