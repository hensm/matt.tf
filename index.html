<!DOCTYPE html>
<html lang="en">
    <head>
        <meta charset="UTF-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <title>matt.tf</title>

        <link rel="stylesheet" href="fonts/stylesheet.css">
        <link rel="stylesheet" href="main.css">

        <script src="main.js" defer></script>
    </head>
    <body>
        <header class="header">
            ===========================<br>
            ========= <div class="header__title">matt.tf</div> =========<br>
            ===========================
        </header>

        <pre id="mail">        ++++++++++
        [
>               +++++++++++
>               ++++++
>               ++++++++++
>               ++++
<<<<            -
        ]

>       - .
>       ++++ .
<       .
>>      --- .
<<      +++++++ .
        .
>>>     ++++++ .
<<<     .
>>      +++++ .</pre>
    </body>
</html>
