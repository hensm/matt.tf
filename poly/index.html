<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="UTF-8">
		<title>poly</title>
		<link rel="stylesheet" href="style.css">
	</head>
	<body>
		<canvas id="canvas"></canvas>

		<script src="utils.js"></script>
		<script src="main.js"></script>
	</body>
</html>