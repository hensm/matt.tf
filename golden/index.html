<!DOCTYPE html>
<html lang="en">
	<head>
		<meta charset="UTF-8">
		<title>Golden</title>
		<link rel="stylesheet" href="main.css">
		<script src="utils.js" defer></script>
		<script src="main.js" defer></script>
	</head>
	<body>
		<div id="controls">
			<div class="controls-container">
				<label for="speed">Speed:</label>
				<input id="speed" type="range" value="35">
			</div>
			<div class="controls-container">
				<label for="color">Color:</label>
				<input id="color" type="color" value="#2dffff">
			</div>
			<div class="controls-container">
				<label for="line-thickness">Line thickness:</label>
				<input id="line-thickness" type="range" value="0">
			</div>
			<div class="controls-container">
				<label for="spiral-thickness">Spiral thickness:</label>
				<input id="spiral-thickness" type="range" value="40">
			</div>
			<div class="controls-container">
				<label for="show-lines">Show lines:</label>
				<input id="show-lines" type="checkbox" checked>
			</div>
			<div class="controls-container">
				<label for="show-spiral">Show spiral:</label>
				<input id="show-spiral" type="checkbox" checked>
			</div>
			<div class="controls-container">
				<label for="counter-clockwise">Draw arcs counter-clockwise:</label>
				<input id="counter-clockwise" type="checkbox">
			</div>
			<div class="controls-container">
				<label for="clear-canvas">Clear each draw:</label>
				<input id="clear-canvas" type="checkbox" checked>
			</div>
		</div>
		<canvas id="canvas"></canvas>
	</body>
</html>